### PR TITLE
[Rakefile] Run Travis tests in parallel

### PR DESCRIPTION
**Pull request by segiddins**
\c @indirect @RochesterinNYC 

this isn't working yet, but I need to study for finals so I thought I'd put this up. the first commit _basically_ works, the second less so

(Might want to use https://github.com/grosser/parallel_tests for the coordination)

### CODE REVIEW

**Comment by segiddins** [comment]
running out of space 😅

File: Rakefile [hunk 1]
```diff
@@ -125,9 +125,14 @@ begin
...
         desc "Run specs with Rubygems #{rg}"
-        RSpec::Core::RakeTask.new(rg) do |t|
-          t.rspec_opts = %w(--format documentation --color)
-          t.ruby_opts  = %w(-w)
+        task rg do
+          patterns = Dir["spec/*"] - %w(spec/spec_helper.rb spec/support)
+          patterns.map do |pattern|
+            pattern = pattern.end_with?(".rb") ? pattern : "#{pattern}/{**/,*/,}*_spec.rb"
+            Thread.new do
+              sh "rspec --pattern '#{pattern}'"
```
Comment (RochesterinNYC) [inline Rakefile]: is this running every single spec in its own thread? Hahah, might have to group specs for a set number of threads less memory or space limits get hit?

**Comment by RochesterinNYC** [comment]
do we want to look into utilizing parallel rspec + separate file systems/dirs to achieve this rather than having every spec in its own thread?

File: Rakefile (hunk 1 above)
Comment (segiddins) [inline Rakefile]: no no, its every subdirectory of specs

**Comment by aledalgrande** [comment]
I would be surprised if this change wouldn't actually slow down tests because the VM keeps swapping. I had more success with tools like Knapsack.

**Comment by segiddins** [comment]
Travis VMs should have enough memory to run multiple ruby processes?

**Comment by aledalgrande** [comment]
Depends on the case, it could probably work for a gem, with Rails apps and fixtures definitely not. Either memory or disk speed was a bottleneck, but probably doesn't apply here: RSpec never goes over 100MB with Bundler.

**Comment by segiddins** [comment]
Closing because this approach didn't work.

**Pull request closed without merging**